### DISCUSSION TITLE
Named Parameter Injection for ChangeUnits does not support jakarta Named

### DISCUSSION THREAD

**Discussion by tmapes** [flamingock/mongock#36] [Ideas]
## Description

Named Parameter Injection for ChangeUnits does not support `jakarta.inject.Named`.

Micronaut 3 moved off the `javax.inject` annotations and instead moved to `jakarta.inject`.

This means for anyone using Named for their ChangeUnits, will have to add back the javax annotations to their project.

## PRIORITY

NORMAL

### Mongock
- Mongock version 5.1.0
- Modules involved(standalone, MongoDB)
- How Mongock is used(builder or annotation approach, etc.) **Builder**

### Environment
- Framework and libraries versions. Especially those that affect directly to Mongock(Spring, Spring data, MongoDB driver, etc.)
- Infrastructure: Kubernetes, Docker, SO, etc.

## Steps to Reproduce

1. Annotate a ChangeUnit injection parameter with `jakarta.inject.Named`

## Behaviour

**Expected behavior:** This `name` is taken into account when searching for a parameter to inject.

**Actual behavior:** The `name` is set to `null` and only the `Class` is taken into account when injecting.
This can lead to issues with multiple dependencies being added to the Runner with the same class. (we found this issue via `MongoCollection<Document>`)

**How often the bug happens:** Every time ChangeUnits are built and executed.

## Link to repository using Mongock

[While not mandatory, accessing to the real code that uses Mongock, would help to solve the issue]

## Additional context

[Any additional information, logs, screenshots, configuration or data that might be necessary to reproduce or understand the issue]

**Comment by dieppa** [comment]
Hi @tmapes, this has been moved to the "suggested ideas" section and will take it to our roadmap. Thanks!
